python/mxnet/executor_manager.py

```diff
@@ -283,8 +283,11 @@ def update_metric(self, metric, labels):
         """ Update evaluation metric with label and current outputs """
         for texec, islice in zip(self.train_execs, self.slices):
             n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))
-            new_slice = slice(islice.start * n, islice.stop * n)
-            labels_slice = [label[new_slice] for label in labels]
+            if n == labels[0].shape[1]:
+                new_slice = slice(islice.start * n, islice.stop * n)
+                labels_slice = [label[new_slice] for label in labels]
+            else:
+                labels_slice = [label[islice] for label in labels]
```

Comment:
The reason for this change is in pull request https://github.com/dmlc/mxnet/issues/2358